libraries/postgres/src/index.js

```diff
@@ -0,0 +1,256 @@
+const {Pool, Client} = require('pg');
+const {dollarQuote} = require('./util');
+const assert = require('assert');
+const debug = require('debug')('taskcluster-lib-postgres');
+
+// Overall plan is: access is only through plpgsql functions, and those must
+// have a consistent API (args + result).  Upgrades can redefine these
+// functions and add new functions, but not change API as existing software may
+// be using those functions concurrently.
+
+const [READ, WRITE] = [Symbol('read'), Symbol('write')];
+
+class Schema{
+  /**
+   * Create a new Schema
+   *
+   * serviceName must be provided, and is used to namespace this service's
+   * tables apart from other services (note that the deployment may put
+   * other services' tables on other database servers entirely, so it is
+   * not allowed to rely on other services' tables)
+   *
+   * script is a script to create the schema, suitable as an argument to
+   * the Postgres DO statment; that is usually 'BEGIN stmt; stmt; .. END'.
+   */
+  constructor({serviceName, script}, {version, versions, methods}={}) {
```

Comment:
I experimented with `{version, versions, methods}={}` - if an empty object is passed in, or an object holding `undefined` in any of the props, or no object was passed in, - in all the cases all these variables are `undefined` in the function. Was that the intended effect?

Reply:
oic. You're avoiding the destructuring type error

Reply:
Yeah, that second positional argument is sort of for "internal use only".  I don't really like this approach (two options-style positional arguments) and would change it before deploying this "for real".